framework/resource/manager.go

```diff
@@ -208,6 +209,23 @@ func (m *Manager) CreateAll(args ...interface{}) error {
 		}
 	}
 
+	// Now that resource state has been filled, populate the declared resource response if available.
+	if m.dcr != nil {
```

Comment:
I could see an argument that it's too magical for ResourceManager to fill in the declared resources at the end of `CreateAll`, and instead there should be a separate `PopulateDeclaredResources` function or something.

Reply:
I don't have strong feelings, but my gut is it's fine here since it's not very complex and only necessitates a couple extra test cases.